File: src/main/java/de/cplaiz/activecraftcore/messages/ColorScheme.java
package de.cplaiz.activecraftcore.messages;

import de.cplaiz.activecraftcore.ActiveCraftPlugin;
import org.bukkit.ChatColor;

public record ColorScheme(ChatColor primary, ChatColor primaryAccent, ChatColor secondary, ChatColor secondaryAccent, ChatColor warningPrefix, ChatColor warningMessage) {
    public static final ColorScheme DEFAULT = new ColorScheme(ChatColor.GOLD, ChatColor.AQUA, ChatColor.GRAY, ChatColor.DARK_GRAY, ChatColor.RED, ChatColor.GRAY);

    public static ColorScheme of(ActiveCraftPlugin plugin) {
        return plugin.getActiveCraftMessage().getColorScheme();
    }
}

File: src/main/java/org/activecraft/activecraftcore/messages/ColorScheme.kt
package org.activecraft.activecraftcore.messages

import org.activecraft.activecraftcore.ActiveCraftPlugin
import org.bukkit.ChatColor

data class ColorScheme constructor(
    @get:JvmName("primary")
    val primary: ChatColor = ChatColor.GOLD,
    @get:JvmName("primaryAccent")
    val primaryAccent: ChatColor = ChatColor.AQUA,
    @get:JvmName("secondary")
    val secondary: ChatColor = ChatColor.GRAY,
    @get:JvmName("secondaryAccent")
    val secondaryAccent: ChatColor = ChatColor.DARK_AQUA,
    @get:JvmName("warningPrefix")
    val warningPrefix: ChatColor = ChatColor.RED,
    @get:JvmName("warningMessage")
    val warningMessage: ChatColor = ChatColor.GRAY
) {

    companion object {
        val default = ColorScheme()
    }

    fun of(plugin: ActiveCraftPlugin): ColorScheme {
        return plugin.activeCraftMessage!!.colorScheme
    }
}